str0m/src/media/mline.rs

```diff
@@ -460,10 +489,76 @@ impl MLine {
 
         Some(NextPacket {
             pt,
-            pkt,
+            seq_no,
             ssrc: pkt.meta.ssrc,
+            body: NextPacketBody::Regular { pkt },
+        })
+    }
+
+    fn create_padding_packet(&mut self, target_size: DataSize, now: Instant) -> Option<NextPacket> {
+        let pt_to_use = {
+            let mut vec: Vec<_> = self
+                .buffers_tx
+                .iter()
+                .map(|(pt, buffer)| (pt, buffer.history_size()))
+                .filter(|(_, size)| *size > 0)
+                .collect();
+            vec.sort_by_key(|(_, size)| *size);
+
+            // Use the last PT i.e. the one that has the most RTX history, this is a poor
+            // approximation for most recent sends.
+            vec.last().map(|(pt, _)| **pt)
+        };
+
+        if let Some(pt_to_use) = pt_to_use {
+            let max_size = target_size * 2_u64;
+
+            let buffer = get_buffer_tx(&self.buffers_tx, pt_to_use)
+                .expect("the buffer to exist as verified previously");
+            let packet = buffer
+                .last_history_packet()
+                .expect("the buffer to have some packet history");
+
+            if packet.data.len() < max_size.as_bytes_usize() {
+                let rtx_pt = self.pt_rtx(pt_to_use)?;
+                let source = match get_source_tx(&mut self.sources_tx, packet.meta.rid, true) {
+                    Some(v) => v,
+                    None => return None,
+                };
+                let seq_no = source.next_seq_no(now);
+
+                return Some(NextPacket {
+                    pt: rtx_pt,
+                    ssrc: source.ssrc(),
+                    seq_no,
+                    body: NextPacketBody::Resend {
+                        pkt: packet,
+                        orig_seq_no: Some(packet.seq_no.expect(
+                            "this packet to have been sent and therefore have a sequence number",
+                        )),
+                    },
+                });
+            }
+        }
+
+        // NB: If we cannot generate padding here for some reason we'll get stuck forever.
+        let pt_to_use = self.buffers_tx.keys().next()?;
```

Comment:
Feels like we should be able to merge this into the above case and not have `if let Some(pt_to_use) = pt_to_use` above.

Reply:
The branches differ not just in the Pt that gets used but also in the kind of padding packet. The first case does a resend of a previous packet, this part generates an empty padding packet

Reply:
Ah yes. Spurious resends :D